contracts/modules/TransactionManager.sol

```diff
@@ -62,44 +64,61 @@ abstract contract TransactionManager is BaseModule {
             address spender = recoverSpender(_wallet, _transactions[i]);
             require(
                 isWhitelisted(_wallet, spender) ||
-                isAuthorised(_wallet, spender, _transactions[i].to, _transactions[i].data),
-                "TM: call not authorised");
+                isAuthorised(_wallet, spender, _transactions[i].to, _transactions[i].data), "TM: call not authorised");
             results[i] = invokeWallet(_wallet, _transactions[i].to, _transactions[i].value, _transactions[i].data);
         }
         return results;
     }
 
     function multiCallWithSession(
         address _wallet,
-        Session calldata,
         Call[] calldata _transactions
     )
         external
         onlySelf()
         onlyWhenUnlocked(_wallet)
         returns (bytes[] memory)
     {
-        bytes[] memory results = new bytes[](_transactions.length);
-        for(uint i = 0; i < _transactions.length; i++) {
-            results[i] = invokeWallet(_wallet, _transactions[i].to, _transactions[i].value, _transactions[i].data);
-        }
-        return results;
+        multiCallWithApproval(_wallet, _transactions);
+    }
+
+    function multiCallWithGuardians(
+        address _wallet,
+        Call[] calldata _transactions
+    )
+        external 
+        onlySelf()
+        onlyWhenUnlocked(_wallet)
+        returns (bytes[] memory)
+    {
+        multiCallWithApproval(_wallet, _transactions);
+    }
+
+    function multiCallWithGuardiansAndStartSession(
+        address _wallet,
+        address _sessionUser,
+        uint64 _duration,
```

Comment:
Would make sense to move the session arguments after the `_transactions` argument to be consistent with the other multicall methods. Also, why not use a `Session` object and have:
```
function multiCallWithGuardiansAndStartSession(
        address _wallet,
        Call[] calldata _transactions,
        Session _session
    )
```

Reply:
Ordering was only because when you just want to start a session, the `transactions` object will be empty and made sense to leave it at the end. I don't mind either way.

Reply:
The `Session` object contains the expiration time whereas here we are passing the duration. They are both uint's so it will work but it's confusing to use these interchangeably.